<a href="https://colab.research.google.com/github/Katerine-Chanci-Echeverri/Project-2---Spaceship-Titanic/blob/main/Proyecto2_Spaceship_Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Proyecto 2 - Parte 1 (base)**

## **Datos personales**

*   **Nombre:** Katerine Chancí Echeverri
*   **ID Coding Dojo:** 80837
*   **Track:** Machine Learning


## **Descripción del problema**

Welcome to the year 2912, where your data science skills are needed to solve a cosmic mystery. We've received a transmission from four lightyears away and things aren't looking good.

The Spaceship Titanic was an interstellar passenger liner launched a month ago. With almost 13,000 passengers on board, the vessel set out on its maiden voyage transporting emigrants from our solar system to three newly habitable exoplanets orbiting nearby stars.

While rounding Alpha Centauri en route to its first destination—the torrid 55 Cancri E—the unwary Spaceship Titanic collided with a spacetime anomaly hidden within a dust cloud. Sadly, it met a similar fate as its namesake from 1000 years before. Though the ship stayed intact, almost half of the passengers were transported to an alternate dimension!

To help rescue crews and retrieve the lost passengers, you are challenged to predict which passengers were transported by the anomaly using records recovered from the spaceship’s damaged computer system.

## **Descripción del conjunto de datos**

| Column                                            | Description                                                                                                                                                                                                                       |
|---------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| PassengerId                                       | A unique Id for each passenger. Each Id takes the form gggg_pp where gggg indicates a group the passenger is travelling with and pp is their number within the group. People in a group are often family members, but not always. |
| HomePlanet                                        | The planet the passenger departed from, typically their planet of permanent residence.                                                                                                                                            |
| CryoSleep                                         | Indicates whether the passenger elected to be put into suspended animation for the duration of the voyage. Passengers in cryosleep are confined to their cabins.                                                                  |
| Cabin                                             | The cabin number where the passenger is staying. Takes the form deck/num/side, where side can be either P for Port or S for Starboard.                                                                                            |
| Destination                                       | The planet the passenger will be debarking to.                                                                                                                                                                                    |
| Age                                               | The age of the passenger.                                                                                                                                                                                                         |
| VIP                                               | Whether the passenger has paid for special VIP service during the voyage.                                                                                                                                                         |
| RoomService, FoodCourt, ShoppingMall, Spa, VRDeck | Amount the passenger has billed at each of the Spaceship Titanic's many luxury amenities.                                                                                                                                         |
| Name                                              | The first and last names of the passenger.                                                                                                                                                                                        |
| Transported                                       | Whether the passenger was transported to another dimension. This is the target, the column you are trying to predict.                                                                                                             |


## **Tarea 1**

Para esta tarea:

Creen un cuaderno de Colab donde hayan subido y mostrado el .head() de cada uno de los conjuntos de datos.  Para cada uno de los conjuntos de datos propuestos, respondan las siguientes preguntas:


1) Fuente de datos.

2) Breve descripción de datos.

3) ¿Cuál es el objetivo?

4) ¿Este es un problema de clasificación o regresión?

5) ¿Cuántas características hay?

6) ¿Cuántas filas de datos hay?

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

## **Solución**


### **Importar librerías**

In [1]:
from google.colab import drive
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.impute import SimpleImputer
from sklearn.compose import make_column_transformer, make_column_selector
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import BaggingClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, precision_score, recall_score, classification_report, ConfusionMatrixDisplay
from sklearn.metrics import confusion_matrix
from sklearn.metrics import roc_auc_score, RocCurveDisplay
from sklearn import set_config
set_config(display='diagram')

### **Cargar el conjunto de datos**

In [2]:
drive.mount("/content/drive")
train = "/content/drive/MyDrive/Colab Notebooks/Coding Dojo 🐉/Proyecto 2/train.csv"
train_df = pd.read_csv(train)
train_df.head()

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


PassengerId HomePlanet CryoSleep  Cabin  Destination   Age    VIP  \
0     0001_01     Europa     False  B/0/P  TRAPPIST-1e  39.0  False   
1     0002_01      Earth     False  F/0/S  TRAPPIST-1e  24.0  False   
2     0003_01     Europa     False  A/0/S  TRAPPIST-1e  58.0   True   
3     0003_02     Europa     False  A/0/S  TRAPPIST-1e  33.0  False   
4     0004_01      Earth     False  F/1/S  TRAPPIST-1e  16.0  False   

   RoomService  FoodCourt  ShoppingMall     Spa  VRDeck               Name  \
0          0.0        0.0           0.0     0.0     0.0    Maham Ofracculy   
1        109.0        9.0          25.0   549.0    44.0       Juanna Vines   
2         43.0     3576.0           0.0  6715.0    49.0      Altark Susent   
3          0.0     1283.0         371.0  3329.0   193.0       Solam Susent   
4        303.0       70.0         151.0   565.0     2.0  Willy Santantines   

   Transported  
0        False  
1         True  
2        False  
3        False  
4         True

In [5]:
train_df.shape

(8693, 14)

In [4]:
test = "/content/drive/MyDrive/Colab Notebooks/Coding Dojo 🐉/Proyecto 2/test.csv"
test_df = pd.read_csv(test)
test_df.head()

PassengerId HomePlanet CryoSleep  Cabin  Destination   Age    VIP  \
0     0013_01      Earth      True  G/3/S  TRAPPIST-1e  27.0  False   
1     0018_01      Earth     False  F/4/S  TRAPPIST-1e  19.0  False   
2     0019_01     Europa      True  C/0/S  55 Cancri e  31.0  False   
3     0021_01     Europa     False  C/1/S  TRAPPIST-1e  38.0  False   
4     0023_01      Earth     False  F/5/S  TRAPPIST-1e  20.0  False   

   RoomService  FoodCourt  ShoppingMall     Spa  VRDeck              Name  
0          0.0        0.0           0.0     0.0     0.0   Nelly Carsoning  
1          0.0        9.0           0.0  2823.0     0.0    Lerome Peckers  
2          0.0        0.0           0.0     0.0     0.0   Sabih Unhearfus  
3          0.0     6652.0           0.0   181.0   585.0  Meratz Caltilter  
4         10.0        0.0         635.0     0.0     0.0   Brence Harperez

### **Preguntas:**

1) Fuente de datos.

El conjunto de datos hace parte del dataset de una de las competencias de Kaggle. Puede encontrarlo [aquí](https://www.kaggle.com/competitions/spaceship-titanic/overview). 

2) Breve descripción de datos. 

Se hace una descripción de los datos al comienzo de este Notebook. 

3) ¿Cuál es el objetivo?

El objetivo es predecir cuáles pasajeros fueron transportados por la anomalía espaciotemporal, usando los registros del sistema de computadora dañado de la nave. 

4) ¿Este es un problema de clasificación o regresión?

Ya que las clases en el vector objetivo son "True" o "False" (variables categóricas), este es un problema de clasificación. 

5) ¿Cuántas características hay?

El conjunto de datos posee 13 características. Esto es, posee 13 columnas, sin contar la columna que será el vector objetivo llamada "Transported".

6) ¿Cuántas filas de datos hay?

En el conjunto de datos de entrenamiento hay 8693 filas. 

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

Los principales retos que preveo para este conjunto de datos es la posibilidad de existencia de datos nulos, la poca colinealidad entre las columnas y el vector objetivo, modelos sobreajustados, existencia de errores de tipo I o II, entre otros. 